Updates and a fix

saiwolf opened the pull request:
Hello!

I saw this repo on my hunt to get Keycloak and Blazor working, and it worked for me, so thank you!

However, I noticed an issue on .net 8 & 9: When you sign out, then sign back in, it "remembers" your user. Further investigation revealed that you are not, in fact, signed out.

If you look at the Blazor Samples repo, Microsoft addresses this by signing out of both Cookie and OIDC providers using a [minimal API endpoint](https://github.com/dotnet/blazor-samples/blob/main/9.0/BlazorWebAppOidcServer/BlazorWebAppOidcServer/LoginLogoutEndpointRouteBuilderExtensions.cs).

Using that sample app repo for reference, I updated this project's files. I also changed the `Login.razor` page to use a form for logout, since the API endpoint uses POST.

Let me know what you think!

saiwolf committed:
Removed Razor Pages in favor of Minimal API auth endpoints. Updated Shared razor components to use the new auth endpoints

saiwolf committed:
Update README to reflect new changes

mattj23 commented:
Hey, this is great, I really appreciate you spending time to improve this project!  When I built it years ago it was under duress, and I've never really had time to do anything but make small fixes as I need them.

Let me give it a try on some of my projects and see how it works.  I may ask some questions if I run into issues but I'll make it a priority to get this merged.

Thanks again!

saiwolf committed:
Converted remaining files to use file-scoped namespace declarations

saiwolf committed:
Refactored code to use primary constructors.

saiwolf committed:
Changed hash implementation in HandleRequirementsAsync to use SHA256.HashData due to CA1850: Prefer static HashData method over ComputeHash

removed 'var' from variables, replaced with definite types.

Refactored IdentityName

saiwolf committed:
Cleanup and refactor Extensions.cs

saiwolf committed:
Added null forgiving ! to keys

saiwolf committed:
Refactor cookie authentication event handling

Moved `OnValidatePrincipal` and `OnTokenValidated` logic
to a new `AuthEventHandlers` class for better organization.

The methods now handle token validation and refresh logic,
simplifying the event handling in the `AddKeycloak` method.

saiwolf committed:
Refactor authentication code for Keycloak integration

Restructure authentication functionality by removing `Extensions.cs` and splitting its logic into `ClaimExtensions.cs` and `KeycloakServiceCollectionExtensions.cs`.

`ClaimExtensions.cs` now manages claims within a `ClaimsIdentity`, while `KeycloakServiceCollectionExtensions.cs` encapsulates the Keycloak authentication setup.

These changes improve code clarity, maintainability, and modularity for future enhancements.

saiwolf committed:
Fixed using directed for AuthEventHandlers

Fixed using directive for KeycloakServiceCollectionExtension

saiwolf committed:
Moved razor components to Components folder.

Moved Auth Endpoints to middleware folder

saiwolf committed:
Organization done

saiwolf committed:
Update README to reflect new refactors
